mono/metadata/w32file-unix.c

```diff
@@ -2331,27 +2334,32 @@ write_file (gint src_fd, gint dest_fd, struct stat *st_src, gboolean report_erro
 	return TRUE ;
 }
 
-#if HAVE_COPYFILE_H
+#if HOST_DARWIN
 static int
-_wapi_copyfile(const char *from, const char *to, copyfile_state_t state, copyfile_flags_t flags)
+_wapi_clonefile(const char *from, const char *to, int flags)
 {
-	gchar *located_from, *located_to;
-	int ret;
+	if (clonefile_ptr != NULL) {
```

Comment:
Since you check `clonefile_ptr ` before calling `_wapi_clonefile`, you can just `g_assert(clonefile_ptr)` here.